packages/core/src/store/level-state-store.ts

```diff
@@ -1,17 +1,20 @@
-import Level from 'level-ts'
+import levelTs from 'level-ts'
+import type Level from 'level-ts'
 import { StreamState, StreamStateHolder, StreamUtils } from '@ceramicnetwork/common'
-import { StateStore } from './state-store'
-import StreamID from '@ceramicnetwork/streamid'
+import { StateStore } from './state-store.js'
+import { StreamID } from '@ceramicnetwork/streamid'
 import * as fs from 'fs'
 import path from 'path'
 
+const LevelC = (levelTs as any).default as unknown as typeof Level
+const defaultLevelFactory = (path: string) => new LevelC(path)
 /**
  * Ceramic store for saving stream state to a local leveldb instance
  */
 export class LevelStateStore implements StateStore {
   #store: Level
 
-  constructor(private storeRoot: string) {}
+  constructor(private storeRoot: string, private readonly levelFactory = defaultLevelFactory) {}
```

Comment:
why do we need this extra level of indirection of a level factory?

Reply:
Added comment there:

> Jest can barely mock modules when running in ESM mode. Some of our tests though rely on `level-ts` package
> being mocked and spied on. Instead of making testing code incredibly ugly, we make dependency on `level-ts` explicit.
> Now we can pass a carefully crafted mock object to spy on.

To be honest, this is a good practice anyway to make such external dependencies somewhat isolated.